### PR TITLE
Mech marketplace integration and staking contract

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

- Add new Mech Marketplace staking contract
- Support new mech marketplace for mech requests/activity
- Update rewards history subgraph to support future contracts
- Fix ability to switch on first run; previously restricted to Pearl Beta
- Fix tenderly lookback limits
- Update trader hash to latest, with fixed betting params & windows support

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by truemiller — APPROVED** [review]

**Review by truemiller — CHANGES_REQUESTED** [review]
pls don't merge yet, need to confirm if there's mech changes

**Review by truemiller — APPROVED** [review]

File: templates/trader.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name: "Trader Agent"
 description: "A single-agent service (sovereign agent) placing bets on Omen"
-hash: bafybeicbfnqfc3potbfmvrmssaxepwilipe5myjbg4xh7ix2sdll4kyc6m
+hash: bafybeidts6y6wytfcwlhnvnacf3vhm6lkiocoe22kkmxxx2onzguez3y3q
 image: https://operate.olas.network/_next/image?url=%2Fimages%2Fprediction-agent.png&w=3840&q=75
-service_version: v0.18.1
+service_version: v0.18.2
```
Comment (jmoreira-valory) [inline templates/trader.yaml:5]: @0xArdi Confirm this version is OK. Version 0.19.0 is for multi-bet strategy, but was downgraded in QS.

File: operate/services/service.py [hunk 2]
```diff
@@ -584,10 +594,17 @@ def _build_host(self, force: bool = True, chain_id: str = "100") -> None:
...
+        # Mech marketplace patch.
+        if "SKILL_TRADER_ABCI_MODELS_PARAMS_ARGS_USE_MECH_MARKETPLACE" in agent_vars:
+            agent_vars[
+                "SKILL_TRADER_ABCI_MODELS_PARAMS_ARGS_USE_MECH_MARKETPLACE"
+            ] = str(chain_data.user_params.use_mech_marketplace).capitalize()
+
+        Path(build, "agent.json").write_text(
+            json.dumps(agent_vars, indent=4),
+            encoding="utf-8",
+        )
```
Comment (jmoreira-valory) [inline operate/services/service.py:607]: @0xArdi Do we need to update further variables?. Same for Docker above.

File: templates/trader.yaml (hunk 1 above)
Comment (jmoreira-valory) [reply templates/trader.yaml:5]: As discussed with @dagacha , version should be v0.18.2

File: frontend/constants/serviceTemplates.ts [hunk 3]
```diff
@@ -4,20 +4,22 @@ import { StakingProgramId } from '@/enums/StakingProgram';
...
   {
     name: 'Trader Agent',
-    hash: 'bafybeicbfnqfc3potbfmvrmssaxepwilipe5myjbg4xh7ix2sdll4kyc6m',
+    hash: 'bafybeidts6y6wytfcwlhnvnacf3vhm6lkiocoe22kkmxxx2onzguez3y3q',
     // hash: 'bafybeibbloa4w33vj4bvdkso7pzk6tr3duvxjpecbx4mur4ix6ehnwb5uu', // temporary
     description: 'Trader agent for omen prediction markets',
     image:
       'https://operate.olas.network/_next/image?url=%2Fimages%2Fprediction-agent.png&w=3840&q=75',
-    service_version: 'v0.18.1',
+    service_version: 'v0.18.2',
```
[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply frontend/constants/serviceTemplates.ts:12]: As discussed with @dagacha , version should be v0.18.2

File: operate/services/manage.py [hunk 4]
```diff
@@ -1362,6 +1389,7 @@ def deploy_service_locally(self, hash: str, force: bool = True) -> Deployment:
         :param force: Remove previous deployment and start a new one.
         :return: Deployment instance
         """
+        self._set_env_variables(hash=hash)
```
Comment (jmoreira-valory) [inline operate/services/manage.py:1392]: @0xArdi Although this approach is far from ideal, I think this is a cleaner, less error-prone way than what we had on [`Deployment._build_host`](https://github.com/valory-xyz/olas-operate-app/blob/main/operate/services/service.py#L581-L590) and [`Deployment._build_docker`](https://github.com/valory-xyz/olas-operate-app/blob/main/operate/services/service.py#L482-L494). 

I wonder why this approach was implemented this way (would the override as environment variable cause any issue?) Let me know any comments.

**Review by iasonrovis — APPROVED** [review]

**Review by 0xArdi — APPROVED** [review]

**Pull request merged**